azkaban-common/src/main/java/azkaban/executor/ExecutorApiGateway.java

```diff
@@ -92,14 +95,15 @@ public ExecutorApiGateway(final ExecutorApiClient apiClient, Props azkProps) {
   }
 
   @VisibleForTesting
-  String createExecutionPath(Optional<Integer> executionId) throws ExecutorManagerException {
+  String createExecutionPath(final Optional<Integer> executionId) throws ExecutorManagerException {
     if (!isReverseProxyEnabled) {
       return "/" + executionResourceName;
     }
 
     if(!executionId.isPresent()) {
-      throw new ExecutorManagerException(
-          "Execution Id must be provided when reverse-proxy is enabled");
+      final String errorMessage = "Execution Id must be provided when reverse-proxy is enabled";
+      logger.error(errorMessage);
```

Comment:
This is totally fine, no need to change again, but one trick that you may want to consider in cases like this is along the lines of:
```suggestion
final ExecutorManagerException eme = new ExecutorManagerException("...");
logger.error(eme);
throw eme;
```
This way you get a stack trace -- that is, the precise location and call chain and maybe the "caused by" if there were other exceptions -- in the log, compared to just a message.

Reply:
Very useful! Thanks for sharing.